src/utils/transaction.ts

```diff
@@ -17,20 +22,62 @@ export type AnyTransaction = Transaction<any, string, string, unknown>;
  **/
 export async function submitTransaction(
 	transaction: AnyTransaction,
-	successMessage?: string
+	successMessage?: string,
+	tryUseMultisig?: Bounty | ChildBounty
 ): Promise<TxFinalizedPayload | undefined> {
+	const proxyAddress = tryUseMultisig?.curator;
+	const multisigAddress = tryUseMultisig?.curatorMultisigAccount;
 	try {
-		const signer = get(polkadotSigner);
+		let signer = get(polkadotSigner);
 		if (!signer) {
 			showErrorDialog('Internal Error, signer is undefined.');
 			return;
 		}
 
-		showLoadingDialog('Submitting transaction');
+		const typedApi = get(dotApi);
+		let calldata = undefined;
+
+		const source = get(activeAccount)?.source;
+
+		// In case it's multisig transaction.
+		if (proxyAddress && multisigAddress && source !== 'mimir') {
+			const matchingMultisig: MultisigInfo | undefined = get(activeAccount)?.multisigs?.find(
+				(multisig) => multisig.address === multisigAddress
+			);
+			if (!matchingMultisig) {
+				showErrorDialog('Internal Error, multisig not found.');
+				return;
+			}
+
+			const multisigSigner = getMultisigSigner(
+				{
+					threshold: matchingMultisig.threshold,
+					signatories: matchingMultisig.signatories
+				},
```

Comment:
```suggestion
				matchingMultisig,
```
Does it have to be explicit or could we simply pass the ready object?